File: backend/demo/src/main/java/com/prospektai/demo/service/PdfSplitter.java
package com.prospektai.demo.service;

import org.apache.pdfbox.multipdf.Splitter;
import org.apache.pdfbox.pdmodel.PDDocument;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

@Component
public class PdfSplitter {

    public List<Path> split(Path source, int pagesPerChunk) throws IOException {
        List<Path> chunkPaths = new ArrayList<>();

        try (PDDocument document = PDDocument.load(source.toFile())) {
            Splitter splitter = new Splitter();
            splitter.setSplitAtPage(pagesPerChunk);

            List<PDDocument> parts = splitter.split(document);

            for (int i = 0; i < parts.size(); i++) {
            PDDocument chunkDoc = parts.get(i);
            Path tempFile = Files.createTempFile("pdf_chunk_" + i + "_", ".pdf");
            try {
                chunkDoc.save(tempFile.toFile());
            } finally {
                chunkDoc.close();
            }
            chunkPaths.add(tempFile);
        }
        }

            return chunkPaths;
        }
}

File: backend/demo/src/main/java/com/prospektai/demo/service/PdfSplitter.kt
package com.prospektai.demo.service

import org.apache.pdfbox.multipdf.Splitter
import org.apache.pdfbox.pdmodel.PDDocument
import org.springframework.stereotype.Component
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

@Component
class PdfSplitter {
    @Throws(IOException::class)
    fun split(source: Path, pagesPerChunk: Int): List<Path> {
        val document = PDDocument.load(source.toFile())
        val splitter = Splitter()
        splitter.setSplitAtPage(pagesPerChunk)

        val parts = splitter.split(document)
        val chunkPaths: MutableList<Path> = ArrayList()

        for (i in parts.indices) {
            val chunkDoc = parts[i]
            val tempFile = Files.createTempFile("pdf_chunk_" + i + "_", ".pdf")
            chunkDoc.save(tempFile.toFile())
            chunkDoc.close()
            chunkPaths.add(tempFile)
        }

        document.close()
        return chunkPaths
    }
}
